be/src/exprs/string_functions.cpp

```diff
@@ -3501,6 +3501,150 @@ StatusOr<ColumnPtr> StringFunctions::regexp_extract_all(FunctionContext* context
     return regexp_extract_all_general(context, options, columns);
 }
 
+Status StringFunctions::regexp_instr_prepare(FunctionContext* context, FunctionContext::FunctionStateScope scope) {
+    // init RE2 options
+    if (scope != FunctionContext::THREAD_LOCAL) {
+        return Status::OK();
+    }
+
+    auto* state = new StringFunctionsState();
+    context->set_function_state(scope, state);
+
+    state->options = std::make_unique<re2::RE2::Options>();
+    state->options->set_log_errors(false);
+    state->options->set_longest_match(false);
+    state->options->set_dot_nl(true);
+
+    // go row regex
+    if (!context->is_notnull_constant_column(1)) {
+        return Status::OK();
+    }
+
+    state->const_pattern = true;
+    auto column = context->get_constant_column(1);
+    auto pattern = ColumnHelper::get_const_value<TYPE_VARCHAR>(column);
+    state->pattern = pattern.to_string();
+    state->regex = std::make_unique<re2::RE2>(state->pattern, *(state->options));
+
+    if (!state->regex->ok()) {
+        std::stringstream error;
+        error << "Invalid regex expression: " << pattern.to_string();
+        context->set_error(error.str().c_str());
+        return Status::InvalidArgument(error.str());
+    }
+
```

Comment:
Take into consideration that the zeroth argument is also constant. for an example
```
select regex_instr('abcdef', 'ab[cd]*') from a_very_large_table
```
or implement constant folding of this function is ScalarOperatorFunctions.java